tests/test_ilastik/test_workflows/testEdgeTrainingWithMulticutGui.py

```diff
@@ -0,0 +1,405 @@
+import logging
+import os
+import platform
+import shutil
+import sys
+import tempfile
+import threading
+import time
+import zipfile
+from pathlib import Path
+
+import h5py
+import numpy
+import pytest
+from elf.evaluation import mean_average_precision
+from PyQt5.QtWidgets import QApplication
+
+from ilastik.applets.dataSelection.opDataSelection import FilesystemDatasetInfo
+from ilastik.workflows.edgeTrainingWithMulticut import EdgeTrainingWithMulticutWorkflow
+from lazyflow.classifiers.parallelVigraRfLazyflowClassifier import ParallelVigraRfLazyflowClassifier
+from lazyflow.utility.timer import Timer
+from tests.test_ilastik.helpers import ShellGuiTestCaseBase
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+logger.setLevel(logging.DEBUG)
+
+
+def waitProcessEvents(timeout=1, event: threading.Event = None):
+    # this is ugly, especially used without an event
+    # there is no guarantee that the state is as expected.
+    # but for now, don't know how else to let gui and all requests settle
+    # for some actions
+    def cond():
+        if event:
+            return event.is_set()
+        else:
+            return False
+
+    start = time.time()
+    while not cond():
+        QApplication.processEvents()
+        if time.time() - start > timeout:
+            return
+
+
+def get_h5_dataset(file_path, dataset_path):
+    with h5py.File(file_path, "r") as f:
+        return f[dataset_path][()]
+
+
+class TestEdgeTrainingWithMulticutGui(ShellGuiTestCaseBase):
+    """Run a set of GUI-based tests on the multicut workflow.
+
+    Note: These tests are named (prefixed with `test_%02d`) in order to impose
+        an order. Tests simulate interaction with a ilastik and depend on
+        the earlier ones.
+    """
+
+    @classmethod
+    def workflowClass(cls):
+        return EdgeTrainingWithMulticutWorkflow
+
+    @classmethod
+    def setup_class(cls):
+        # Base class first
+        super().setup_class()
+
+        # input files:
+        current_dir = Path(__file__).parent
+        cls.sample_data_raw = current_dir.parent / "data" / "inputdata" / "3d1c-synthetic.h5"
+        cls.sample_data_probs = current_dir.parent / "data" / "inputdata" / "3d1c-synthetic_Probabilities.h5"
+
+        # output files:
+        cls.temp_dir = Path(tempfile.mkdtemp())
```

Comment:
Any reason not to use pytest's [`tmp_path`](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) fixture?

Reply:
hm I didn't want to dig into the workarounds to make it usable in the setup class method.